metadataStore/odm_templates.py

```diff
@@ -26,79 +27,111 @@ class BeginRunEvent(Document):
+        Globally unique id for this run
     time : timestamp
-        The date/time as found at the client side when an event is
+        The date/time as found at the client side when an begin_run_event is
         created.
-    beamline_id: str
+    beamline_id : str
         Beamline String identifier. Not unique, just an indicator of
         beamline code for multiple beamline systems
-    owner: str, optional
+    project : str, optional
+        Name of project that this run is part of
+    owner : str
         Specifies the unix user credentials of the user creating the entry
-    scan_id : int, optional
+    group : str
+        Unix group to associate this data with
+    scan_id : int
          scan identifier visible to the user and data analysis
-    beamline_config: bson.ObjectId, optional
+    beamline_config: bson.ObjectId
         Foreign key to beamline config corresponding to a given run
-    custom: dict, optional
-        Additional parameters that data acquisition code/user wants to
-        append to a given header. Name/value pairs
+    sample : dict
+        Information about the sample, may be a UID to another collection
     """
+    uid = StringField(required=True, unique=True)
     time = FloatField(required=True)
-    time_as_datetime = DateTimeField()
+    time_as_datetime = DateTimeField(required=False)
+    project = StringField(required=False)
     beamline_id = StringField(max_length=20, unique=False, required=True)
-    scan_id = StringField(required=False, unique=False)
+    scan_id = IntField(required=True)
     beamline_config = ReferenceField(BeamlineConfig, reverse_delete_rule=DENY,
-                                     required=False,
+                                     required=True,
                                      db_field='beamline_config_id')
-    # one-to-many relationship constructed between BeginRunEvent and BeamlineConfig
     owner = StringField(default=getuser(), required=True, unique=False)
-    custom = DictField(unique=False, required=False, default=dict())
-    # Keeping custom a dict for the time being instead of new collection
-
-    meta = {'indexes': ['-_id', '-owner', '-time']}
+    group = StringField(required=False, unique=False, default=None)
+    sample = DictField(required=False)  # lightweight sample placeholder.
+    meta = {'indexes': ['-_id', '-owner', '-time', '-scan_id', '-uid']}
 
 
-class EndRunEvent(Document):
+class EndRunEvent(DynamicDocument):
     """Indicates the end of a series of events
 
     Attributes
     ----------
     begin_run_event : bson.ObjectId
         Foreign key to corresponding BeginRunEvent
-    reason : str
+    exit_status : {'success', 'fail', 'abort'}
         provides information regarding the run success.
-    time : timestamp
+    time : float
         The date/time as found at the client side when an event is
         created.
+    reason : str, optional
+        Long-form description of why the run ended.
+        20 char max.
```

Comment:
Did this 20 character limit come from somewhere? or just an arbitrary limit imposed by (somewhat) BDFL Arkilic?